File: ai-workspace/workspace-prompt/src/main/java/com/aiworkspace/prompt/entity/Prompt.java
package com.aiworkspace.prompt.entity;

import com.aiworkspace.common.core.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 提示词实体
 *
 * 对应 prompt 表，表示用户私有的提示词模板；通过 createBy 隔离归属，
 * 支持按标题关键字与分类检索。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("prompt")
public class Prompt extends BaseEntity {

    private String title;

    /** 提示词模板正文，可含占位符供用户填写后复用到 Chat */
    private String content;

    /** 自由分类标签，用于按场景分组（如"写作"、"编程"、"翻译"）；可为空 */
    private String category;

    private String description;

    /** 归属用户ID（资源隔离列），按 createBy 隔离避免越权访问 */
    private Long createBy;
}

File: ai-workspace-backend/internal/model/prompt.go
package model

// Prompt 提示词模板，对应 prompt 表。
// 用户私有，按 CreateBy 隔离；支持按标题关键字与分类筛选，可复用到 Chat。
type Prompt struct {
	UserOwnedModel
	Title       string `gorm:"column:title;size:200"        json:"title"`
	// 提示词正文，可含占位符供用户填写后使用
	Content     string `gorm:"column:content;type:text"     json:"content"`
	// 自由分类标签，如"写作"、"编程"、"翻译"；可为空
	Category    string `gorm:"column:category;size:100"     json:"category"`
	Description string `gorm:"column:description;size:500"  json:"description"`
}

func (Prompt) TableName() string { return "prompt" }
